[CI] RepositoryVerifyIntegrityIT testTransportException failing

**Build Scans:**
- [elasticsearch-intake #9591 / part2](https://gradle-enterprise.elastic.co/s/xkyt5pb6w2bgw)
- [elasticsearch-periodic-platform-support #3995 / sles-15_platform-support-unix](https://gradle-enterprise.elastic.co/s/sladcbhkhkhpy)

**Reproduction Line:**
```
./gradlew ":x-pack:plugin:snapshot-repo-test-kit:internalClusterTest" --tests "org.elasticsearch.repositories.blobstore.testkit.integrity.RepositoryVerifyIntegrityIT.testTransportException" -Dtests.seed=4B676F2F3738A1A3 -Dtests.locale=gsw-LI -Dtests.timezone=Australia/Adelaide -Druntime.java=22
```

**Applicable branches:**
main

**Reproduces locally?:**
N/A

**Failure History:**
[See dashboard](https://es-delivery-stats.elastic.dev/app/dashboards#/view/dcec9e60-72ac-11ee-8f39-55975ded9e63?_g=(refreshInterval:(pause:!t,value:60000),time:(from:now-7d%2Fd,to:now))&_a=(controlGroupInput:(chainingSystem:HIERARCHICAL,controlStyle:twoLine,ignoreParentSettings:(ignoreFilters:!f,ignoreQuery:!f,ignoreTimerange:!f,ignoreValidations:!t),panels:('0c0c9cb8-ccd2-45c6-9b13-96bac4abc542':(explicitInput:(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,enhancements:(),fieldName:task.keyword,grow:!t,id:'0c0c9cb8-ccd2-45c6-9b13-96bac4abc542',searchTechnique:wildcard,selectedOptions:!(),singleSelect:!t,title:'Gradle%20Task',width:medium),grow:!t,order:0,type:optionsListControl,width:small),'144933da-5c1b-4257-a969-7f43455a7901':(explicitInput:(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,enhancements:(),fieldName:name.keyword,grow:!t,id:'144933da-5c1b-4257-a969-7f43455a7901',searchTechnique:wildcard,selectedOptions:!('testTransportException'),title:Test,width:medium),grow:!t,order:2,type:optionsListControl,width:medium),'4e6ad9d6-6fdc-4fcc-bf1a-aa6ca79e0850':(explicitInput:(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,enhancements:(),fieldName:className.keyword,grow:!t,id:'4e6ad9d6-6fdc-4fcc-bf1a-aa6ca79e0850',searchTechnique:wildcard,selectedOptions:!('org.elasticsearch.repositories.blobstore.testkit.integrity.RepositoryVerifyIntegrityIT'),title:Suite,width:medium),grow:!t,order:1,type:optionsListControl,width:medium)))))

**Failure Message:**
```
java.lang.AssertionError: org.elasticsearch.common.util.CancellableThreads$ExecutionCancelledException: operation was cancelled reason [task cancelled]
```

**Issue Reasons:**
- [main] 2 failures in test testTransportException (2.6% fail rate in 77 executions)

**Note:**
This issue was created using new test triage automation. Please report issues or feedback to es-delivery.

--- a/muted-tests.yml
+++ b/muted-tests.yml
@@ -211,9 +211,6 @@ tests:
 - class: org.elasticsearch.repositories.blobstore.testkit.integrity.RepositoryVerifyIntegrityIT
   method: testCorruption
   issue: https://github.com/elastic/elasticsearch/issues/112769
-- class: org.elasticsearch.repositories.blobstore.testkit.integrity.RepositoryVerifyIntegrityIT
-  method: testTransportException
-  issue: https://github.com/elastic/elasticsearch/issues/112779
 
 # Examples:
 #

--- a/x-pack/plugin/snapshot-repo-test-kit/src/main/java/org/elasticsearch/repositories/blobstore/testkit/integrity/RepositoryIntegrityVerifier.java
+++ b/x-pack/plugin/snapshot-repo-test-kit/src/main/java/org/elasticsearch/repositories/blobstore/testkit/integrity/RepositoryIntegrityVerifier.java
@@ -934,7 +934,11 @@ class RepositoryIntegrityVerifier {
                         if (cancellableThreads.isCancelled()) {
                             runnable.onFailure(new TaskCancelledException("task cancelled"));
                         } else {
-                            cancellableThreads.execute(runnable::run);
+                            try {
+                                cancellableThreads.execute(runnable::run);
+                            } catch (RuntimeException e) {
+                                runnable.onFailure(e);
+                            }
                         }
                     }
                 }

